src/app/AppNavItem.vue

```diff
@@ -113,33 +113,24 @@ const isActive = computed(() => {
     return false
   }
 
-  if (props.routeName === currentRoute.name) {
+  if (props.routeName === route.name) {
     return true
   }
 
-  const currentRouteSubpath = currentRoute.path.split('/')[2]
+  const currentRouteSubpath = route.path.split('/')[2]
   if (currentRouteSubpath === targetRoute.value.name) {
     return true
   }
 
-  if (currentRoute.meta.parent) {
-    try {
-      const parentRoute = router.resolve({ name: currentRoute.meta.parent })
-
-      if (parentRoute.name === props.routeName) {
-        return true
-      }
-    } catch (error) {
-      if (error instanceof Error && error.message.includes('No match for')) {
-        // Unfortunately, `router.resolve` throws instead of returning `null` when a route can’t be resolved so we ignore this particular error because we don’t care about non-existing routes here.
-        console.warn(error)
-      } else {
-        throw error
-      }
-    }
-  }
+  try {
+    return props.routeName && route.matched.some((matchedRoute) => {
+      const resolvedRoute = router.resolve(matchedRoute)
 
-  return props.routeName && currentRoute.matched.some((route) => props.routeName === route.name || props.routeName === route.redirect)
+      return props.routeName === resolvedRoute.name
+    })
+  } catch {
+    return false
```

Comment:
Is not having something visible in console here likely to be a gotcha if we make a mistake and are trying to figure out why something isn't working? I'm not sure if after this change a console error would still surface - if it does you can ignore this.

Reply:
You’re right, we should probably log errors here to have better visibility on what’s likely an unintentional error case. I’ve added `console.error` statement.